PhysicsTools/PyTorchAlpaka/interface/Policy.h

```diff
@@ -0,0 +1,259 @@
+#ifndef PhysicsTools_PyTorchAlpaka_interface_Policy_h
+#define PhysicsTools_PyTorchAlpaka_interface_Policy_h
+
+#include <cstddef>
+#include <optional>
+#include <type_traits>
+#include <cassert>
+
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/memory.h"
+
+namespace cms::torch::alpakatools {
+
+  template <typename T>
+  using HostBuffer = cms::alpakatools::host_buffer<T[]>;
+  template <typename TDevice, typename T>
+  using DeviceBuffer = cms::alpakatools::device_buffer<TDevice, T[]>;
+
+  enum class MemcpyKind : uint8_t {
+    // Copy data from host to device (used primarily for ROCm/HIP backends)
+    HostToDevice = 0,
+    // Copy data from device to host (used for CPU fallback inference)
+    DeviceToHost = 1,
+    // Copy data between device memory regions (e.g. for constant data)
+    DeviceToDevice = 2,
+    // Copy data between host memory regions (for CPU semantics, maps to DeviceToDevice in alpaka world).
+    HostToHost = 2
+  };
+
+  template <typename TDev, typename T>
+  class Policy;
+
+#ifdef ALPAKA_ACC_GPU_CUDA_ENABLED
+
+  // Provides D2D (Device-to-Device) copy support for constant data only.
+  // Non-const tensors directly reuse the provided device pointer.
+  template <typename T>
+  class Policy<alpaka_cuda_async::Device, T> {
+  public:
+    using Ttype = std::remove_const_t<T>;
+
+    explicit Policy(T* data_ptr, const size_t num_elems) : num_elems_(num_elems), data_ptr_(data_ptr) {}
+
+    // Perform a copy operation for constant tensors.
+    // Only `MemcpyKind::DeviceToDevice` is supported.
+    // Non-const tensors reuse the provided pointer.
+    void copy(void* queue_ptr, const MemcpyKind kind) {
```

Comment:
I think we need a very strong justification for introducing a new interface based on `void*`.

Since `Policy` is specialised for the different backends, can it use the explicit queue type ?